src/dodal/utils.py

```diff
@@ -202,7 +268,14 @@ def invoke_factories(
         dependent_name = leaves.pop()
         params = {name: devices[name] for name in dependencies[dependent_name]}
         try:
-            devices[dependent_name] = factories[dependent_name](**params, **kwargs)
+            factory = factories[dependent_name]
+            if isinstance(factory, DeviceInitializationController):
+                mock = kwargs.get("mock", False) or kwargs.get(
+                    "fake_with_ophyd_sim", False
+                )
```

Comment:
Could: Is `fake_with_ophyd_sim` purely in there for backwards compatibility downstream? I think from an MX POV we can break compatibility here and it will be fine, we will need to do a bit of migration work anyway.

Reply:
Yes that's there for backwards compat- is dodal still wanting a "make all devices" method (just for `dodal connect`?) or is that API going to move into blueapi? I was going to make a PR after this got merged to remove the kwargs and pass explicitly the args that connect wants (after moving more beamlines over).

Reply:
I'm also happy to break compatibility, however this change is required to fix some ongoing blueapi build issues so maybe we shouldn't block it/do that separately?

https://github.com/DiamondLightSource/blueapi/issues/665
https://github.com/DiamondLightSource/blueapi/issues/647

Reply:
Yes, we still want `make_all_devices` I think. If it's going to be addressed in another issue that's fine.

Reply:
https://github.com/DiamondLightSource/dodal/issues/844